### PR TITLE
Acceptance tests for actions performed on received shares

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add acceptance tests for actions performed on received/offered shares.All the scenarios that have  been implemented in these PR:#2292 and #2290 have also been deleted to get rid of duplication

## Related Issue

- Fixes issue #2478

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by jasson99** [comment]
Please review : @dpakach @skshetry @HariBhandari07 @kiranparajuli589

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature [hunk 1]
```diff
@@ -8,239 +8,142 @@ Feature: accept/decline shares coming from internal users
       | username |
       | user1    |
       | user2    |
+      | user3    |
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]: It's better to create a user in individual scenario if the user is not used in every scenario.

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature [hunk 2]
```diff
@@ -8,239 +8,142 @@ Feature: accept/decline shares coming from internal users
...
-  @skip @yetToImplement
-  Scenario: receive shares with same name from different users, accept one by one
-    Given the setting "Automatically accept new incoming local user shares" in the section "Sharing" has been disabled
-    And user "user2" has created folder "/simple-folder/from_user2"
-    And user "user2" has shared folder "/simple-folder" with user "user3"
-    And user "user1" has created folder "/simple-folder/from_user1"
-    And user "user1" has shared folder "/simple-folder" with user "user3"
-    And user "user3" has logged in using the webUI
-    When the user accepts share "simple-folder" offered by user "User One" using the webUI
-    And the user accepts share "simple-folder" offered by user "User Two" using the webUI
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]: Why did you delete all these scenarios? Are they already implemented in other feature files?

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature [hunk 3]
```diff
@@ -8,239 +8,142 @@ Feature: accept/decline shares coming from internal users
...
-    And user "user2" has shared folder "/simple-folder" with group "grp1"
-    And user "user1" has logged in using the webUI
-    When the user declines share "simple-folder" offered by user "User Two" using the webUI
-    And the user reloads the current page of the webUI
-    Then folder "simple-folder" should be in state "Declined" in the shared-with-you page on the webUI
-    And folder "simple-folder (2)" should not be listed in the files page on the webUI
+    Given the setting "shareapi_auto_accept_share" of app "core" has been set to "no"
+    And these groups have been created:
+      | groupname |
+      | grp1  |
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]: Use ```Ctrl + Alt + L``` to correctly format this

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 4]
```diff
@@ -648,6 +649,10 @@ Given('user {string} has declined the share {string} offered by user {string}',
   return sharingHelper.declineShare(filename, user, sharer)
 })
 
+Given('user {string} has accepted the share {string} offered by user {string}', async function (user, filename, sharer) {
```
Comment (HariBhandari07) [inline tests/acceptance/stepDefinitions/sharingContext.js]: Why are you using async function here?

File: tests/acceptance/helpers/sharingHelper.js [hunk 5]
```diff
@@ -167,5 +167,23 @@ module.exports = {
           })
       }
     }
+  },
+  acceptShare: async function (filename, user, sharer) {
```
Comment (HariBhandari07) [inline tests/acceptance/helpers/sharingHelper.js]: Documentation please

**Comment by kiranparajuli589** [comment]
LGTM :+1: 
IMO single commit would be better!

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature (hunk 2 above)
Comment (jasson99) [reply tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]: yes they have been implemented already

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature (hunk 1 above)
Comment (jasson99) [reply tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]: okay..ThankYou

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6537/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6537/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6541/

<p>





</p></details>

File: tests/acceptance/helpers/sharingHelper.js [hunk 6]
```diff
@@ -185,5 +193,32 @@ module.exports = {
...
+        const shareID = element.id
+        const headers = httpHelper.createAuthHeader(user)
+        const apiURL = client.globals.backend_url + '/ocs/v2.php/apps/files_sharing/api/v1/shares/pending/' + shareID + '?format=json'
+        return fetch(apiURL,
+          {
+            method: 'POST',
+            headers: headers
+          })
+          .then(res => {
+            httpHelper.checkStatus(res, 'The response status is not the expected value')
```
Comment (skshetry) [inline tests/acceptance/helpers/sharingHelper.js]: Also check OCS status code.

File: tests/acceptance/helpers/sharingHelper.js [hunk 7]
```diff
@@ -185,5 +193,32 @@ module.exports = {
...
+   *
+   * @async
+   * @param {string} filename
+   * @param {string} user
+   * @param {string} sharer
+   */
+  acceptShare: async function (filename, user, sharer) {
+    const allShares = await this.getAllShares(user)
+    for (const element of allShares) {
+      if (element.state === 1 && element.path.replace(/^\/|\/$/g, '') === filename && element.uid_owner === sharer) {
```
Comment (skshetry) [inline tests/acceptance/helpers/sharingHelper.js]: Don't use magic number

File: tests/acceptance/helpers/sharingHelper.js [hunk 8]
```diff
@@ -185,5 +193,32 @@ module.exports = {
...
+        return fetch(apiURL,
+          {
+            method: 'POST',
+            headers: headers
+          })
+          .then(res => {
+            httpHelper.checkStatus(res, 'The response status is not the expected value')
+          })
+      }
+    }
```
Comment (skshetry) [inline tests/acceptance/helpers/sharingHelper.js]: Throw exception if it does not find specified share

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingAcceptShares</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6566/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6566/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>iPhone</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6566/

<p>





</p></details>

**Comment by jasson99** [comment]
@skshetry

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6568/

<p>





</p></details>

File: tests/acceptance/helpers/sharingHelper.js [hunk 9]
```diff
@@ -168,22 +172,72 @@ module.exports = {
...
-        return fetch(apiURL,
-          {
-            method: 'DELETE',
-            headers: headers
-          })
-          .then(res => {
-            httpHelper.checkStatus(res, 'The response status is not the expected value')
-          })
-      }
+    const elementToDecline = allShares.filter((element) => {
```
Comment (skshetry) [inline tests/acceptance/helpers/sharingHelper.js]:
```suggestion
    const elementsToDecline = allShares.filter((element) => {
```

File: tests/acceptance/helpers/sharingHelper.js [hunk 10]
```diff
@@ -168,22 +172,72 @@ module.exports = {
...
+   * Asynchronously accepts the shares in pending state and meeting the conditions
+   *
+   * @async
+   * @param {string} filename
+   * @param {string} user
+   * @param {string} sharer
+   */
+  acceptShare: async function (filename, user, sharer) {
+    const allShares = await this.getAllSharesSharedWithUser(user)
+    const elementToAccept = allShares.filter((element) => {
```
Comment (skshetry) [inline tests/acceptance/helpers/sharingHelper.js]:
```suggestion
    const elementsToAccept = allShares.filter((element) => {
```

File: tests/acceptance/helpers/sharingHelper.js [hunk 11]
```diff
@@ -18,6 +18,11 @@ module.exports = {
     share: 16,
     all: 31
   }),
+  FILE_STATE: Object.freeze({
```
Comment (dpakach) [inline tests/acceptance/helpers/sharingHelper.js]:
```suggestion
  SHARE_STATE: Object.freeze({
```

**Comment by dpakach** [comment]
Looks good in general but please make sure the scenarios that you deleted are already implemented somewhere.

**Comment by jasson99** [comment]
@individual-it

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature [hunk 12]
```diff
@@ -10,237 +10,139 @@ Feature: accept/decline shares coming from internal users
...
-    And the user shares folder "simple-folder (2)" with group "grp1" using the webUI
-    And the user declines share "simple-folder (2)" offered by user "User Two" using the webUI
-    And the user reloads the current page of the webUI
-    Then folder "simple-folder (2)" should be in state "Declined" in the shared-with-you page on the webUI
-    And folder "simple-folder (2)" should not be listed in the files page on the webUI
+    When the user shares folder "simple-folder (2)" with group "grp1" as "Viewer" using the webUI
+    And the user unshares folder "simple-folder (2)" using the webUI
+    And the user browses to the shared-with-me page using the webUI
+    Then folder "simple-folder" shared by "User One" should be in "Declined" state on the webUI
+    And folder "simple-folder" shared by "User Two" should be in "Pending" state on the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]: that share should not be listed here. I've opened an issue #2512
please mark the test with this issue

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6602/

<p>





</p></details>

File: tests/acceptance/pageObjects/sharedWithMePage.js [hunk 13]
```diff
@@ -82,6 +82,14 @@ module.exports = {
         locateStrategy: this.elements.getSharedFromUserName.locateStrategy,
         selector: requiredXpath
       })
+    },
+    assertShareIsAbsent: function (element, sharer) {
```
Comment (skshetry) [inline tests/acceptance/pageObjects/sharedWithMePage.js]: We decided to not have assert on page. Considering this PR to be small, I guess, you could just make one function that returns all shares, and assert on the context.

**Comment by jasson99** [comment]
@skshetry @individual-it

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 14]
```diff
@@ -771,3 +771,10 @@ Then('the user should be redirected to the files-drop page', async function () {
 Then('the user should be redirected to the public links page', async function () {
   return client.page.publicLinkFilesPage().waitForPage()
 })
+
+Then('file/folder {string} shared by {string} should not be listed in the webUI', function (element, sharer) {
+  return client.page.sharedWithMePage().isSharePresent(element, sharer)
+    .then(found => {
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/filesContext.js]: Use `async`/`await` here

**Comment by skshetry** [comment]
Please squash all the commits.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6687/

<p>





























</p></details>
